cli/doc/tests.rs

```diff
@@ -41,886 +42,267 @@ impl DocFileLoader for TestLoader {
   }
 }
 
-#[tokio::test]
-async fn export_fn() {
-  let source_code = r#"/**
-* @module foo
-*/
-
-/**
-* Hello there, this is a multiline JSdoc.
-*
-* It has many lines
-*
-* Or not that many?
-*/
-export function foo(a: string, b?: number, cb: (...cbArgs: unknown[]) => void, ...args: unknown[]): void {
-    /**
-     * @todo document all the things.
-     */
-    console.log("Hello world");
+macro_rules! doc_test {
+  ( $name:ident, $source:expr; $block:block ) => {
+    doc_test!($name, $source, false, false; $block);
+  };
+
+  ( $name:ident, $source:expr, details; $block:block ) => {
+    doc_test!($name, $source, true, false; $block);
+  };
+
+  ( $name:ident, $source:expr, private; $block:block ) => {
+    doc_test!($name, $source, false, true; $block);
+  };
+
+  ( $name:ident, $source:expr, details, private; $block:block ) => {
+    doc_test!($name, $source, true, true; $block);
+  };
+
+  ( $name:ident, $source:expr, $details:expr, $private:expr; $block:block ) => {
+    #[tokio::test]
+    async fn $name() {
+      let source_code = $source;
+      let details = $details;
+      let private = $private;
+
+      let loader =
+        TestLoader::new(vec![("test.ts".to_string(), source_code.to_string())]);
+      let entries = DocParser::new(loader, private)
+        .parse("test.ts")
+        .await
+        .unwrap();
+
+      let doc = DocPrinter::new(&entries, details, private).to_string();
+      #[allow(unused_variables)]
```

Comment:
Why is this supression needed?

Reply:
`json_test!`s don't use the doc variable, so in those tests it would then have doc unused.